images/handlers_test.go

```diff
@@ -0,0 +1,372 @@
+)
+
+const (
+	tarFileLength = (4 << 10)
+
+	imageLayoutVersion = "1.0.0"
+
+	configStr = `{
+    "created": "2015-10-31T22:22:56.015925234Z",
+    "author": "Alyssa P. Hacker <alyspdev@example.com>",
+    "architecture": "amd64",
+    "os": "linux",
+    "config": {
+        "User": "alice",
+        "ExposedPorts": {
+            "8080/tcp": {}
+        },
+        "Env": [
+            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+            "FOO=oci_is_a",
+            "BAR=well_written_spec"
+        ],
+        "Entrypoint": [
+            "/bin/my-app-binary"
+        ],
+        "Cmd": [
+            "--foreground",
+            "--config",
+            "/etc/my-app.d/default.cfg"
+        ],
+        "Volumes": {
+            "/var/job-result-data": {},
+            "/var/log/my-app-logs": {}
+        },
+        "WorkingDir": "/home/alice",
+        "Labels": {
+            "com.example.project.git.url": "https://example.com/project.git",
+            "com.example.project.git.commit": "45a939b2999782a3f005621a8d0f29aa387e1d6b"
+        }
+    },
+    "rootfs": {
+      "diff_ids": [
+        "",
+        ""
+      ],
+      "type": "layers"
+    },
+    "history": [
+      {
+        "created": "2015-10-31T22:22:54.690851953Z",
+        "created_by": "/bin/sh -c #(nop) ADD file:a3bc1e842b69636f9df5256c49c5374fb4eef1e281fe3f282c65fb853ee171c5 in /"
+      },
+      {
+        "created": "2015-10-31T22:22:55.613815829Z",
+        "created_by": "/bin/sh -c #(nop) CMD [\"sh\"]",
+        "empty_layer": true
+      }
+    ]
+}
+`
+)
+
+var (
+	configDesc ocispec.Descriptor
+
+	expDiffIDs = []digest.Digest{
+		"sha256:c6f988f4874bb0add23a778f753c65efe992244e148a1d2ec2a8b664fb66bbd1",
+		"sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef",
+	}
+)
+
+type checker interface {
+	Fatal(args ...interface{})
+}
+
+type tarContent struct {
+	header *tar.Header
+	b      []byte
+}
+
+type imageLayout struct {
+	rootDir string
+	config  string
+	tarList []tarContent
+}
+
+func TestChildrenHandler(t *testing.T) {
+	var exp []ocispec.Descriptor
+	ctx, _, store, manifest, config, layers, cleanup := setupImageStore(t)
+	defer cleanup()
+
+	handler := ChildrenHandler(store)
+	subDescs, err := handler(ctx, manifest)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	exp = append(exp, config)
+	exp = append(exp, layers...)
+	if !reflect.DeepEqual(subDescs, exp) {
+		t.Fatalf("descriptors[%+v] not match to the expected[%+v]!", subDescs, exp)
+	}
+}
+
+func setupImageStore(t checker) (ctx context.Context, root string, store content.Store,
+	manifest, config ocispec.Descriptor, layers []ocispec.Descriptor,
+	clean func()) {
+
+	var (
+		imageRuntime ocispec.Image
+		contentList  []tarContent
+	)
+
+	root, err := ioutil.TempDir("", "oci-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create image layout directory
+	err = os.MkdirAll(filepath.Join(root, "blobs", "sha256"), 0700)
```

Comment:
If you are going to use the existing content store, just initialize one with a root and use the content store interfaces for populating the content. This approach is too brittle as it is relying on the internal structure of the content store.